src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -688,8 +688,8 @@ public function findMany($ids, $columns = ['*'])
             return $this->getRelated()->newCollection();
         }
 
-        return $this->whereIn(
-            $this->getRelated()->getQualifiedKeyName(), $this->parseIds($ids)
+        return $this->getRelated()->whereKey(
```

Comment:
This looks like different logic to me. Call `whereKey` on `this`.

Reply:
I wonder how the tests pass in both cases, but I thought the key name should be derived based on the related model.

Reply:
I wonder how the tests pass in both cases, I thought that the key name should be derived from the related model, since we had ` $this->getRelated()->getQualifiedKeyName()` before, as the first argument to whereIn.